hip/matrix/dense_kernels.hip.cpp

```diff
@@ -81,15 +81,20 @@ void simple_apply(std::shared_ptr<const HipExecutor> exec,
 {
     if (hipblas::is_supported<ValueType>::value) {
         auto handle = exec->get_hipblas_handle();
-        {
-            hipblas::pointer_mode_guard pm_guard(handle);
-            auto alpha = one<ValueType>();
-            auto beta = zero<ValueType>();
-            hipblas::gemm(handle, HIPBLAS_OP_N, HIPBLAS_OP_N, c->get_size()[1],
-                          c->get_size()[0], a->get_size()[1], &alpha,
-                          b->get_const_values(), b->get_stride(),
-                          a->get_const_values(), a->get_stride(), &beta,
-                          c->get_values(), c->get_stride());
+        if (c->get_size()[0] * c->get_size()[1] > 0) {
+            if (a->get_size()[1] > 0) {
```

Comment:
should we allow the empty matrix do some operation?

Reply:
Not doing it would be more surprising IMO, just like it's surprising that CUDA throws an error on empty kernel launches. Avoiding behavioral edge cases in your API is always good practice.